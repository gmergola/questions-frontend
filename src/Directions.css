.Directions-container {
  margin-left: 80px;
  margin-right: 80px;
  margin-bottom: 50%;
}

.Directions-home-button {
  margin-top: 15px;
  margin-bottom: 30px;
  border-radius: 20px;
  width: 150px;
  height: 30px;
  border: none;
}

.Directions-bullet-point {
  margin-bottom: 40px;
}

.dot {
  margin-right: 15px;
  height: 10px;
  width: 20px;
  background-color: rgb(250, 204, 119);
  border-radius: 50%;
  display: inline-block;
}

.Directions-home-button:hover {
  opacity: 0.5;
}

@media only screen and (max-width: 600px){
  .Directions-container {
    margin-left: 30px;
    margin-right: 30px;
  }
}