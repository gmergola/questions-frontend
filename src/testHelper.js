import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

/**
 * Helper function to modify the render function in React Testing Library
 * by wrapping rendered components inside of a router.
 *
 * Taken from https://testing-library.com/docs/example-react-router
 *
 * @param {Component} component - React component we want to render
 * @param {Object} param1 - object with route and history keys
 */

export function renderWithRouter(component,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...render(component, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  };
}