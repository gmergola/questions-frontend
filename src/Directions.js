import React from 'react';
import { useHistory } from "react-router-dom";
import './Directions.css';

/**Directions: a component with a route of /directions that explains to the user how to play the game */
function Directions() {
  const history = useHistory();
  return (
    <div className="Directions-container">
      <button
        className="Directions-home-button"
        onClick={() => history.push('/')}>
        <b>Back Home</b>
      </button>
      <div>
        <h3>A Family Feud inspired game with a twist!</h3>
        <div className="Directions-bullet-point">
          <span className="dot"></span><span>Select a question</span>
        </div>

        <div className="Directions-bullet-point">
          <span className="dot"></span><span>
            Once you've selected a question decide who is on team1 and who is on team 2.
          </span>
        </div>

        <div className="Directions-bullet-point">
          <span className="dot"></span><span>
            Once you've decided on the teams, have team1 and team 2 each submit an answer.
            After that, the team with the highest voted answer starts!
          </span>
        </div>

        <div className="Directions-bullet-point">
          <span className="dot"></span><span>
            Once the starting team has been selected, that team has three chances to guess
            as many correct answers as they can. If that team hits three incorrect answers, it is
            the other team's turn.
          </span>
        </div>

        <div className="Directions-bullet-point">
          <span className="dot"></span><span>
            Every question has 6 answers, when all 6 correct answers have been guessed, the team with
            the highest remaining points wins that round!
          </span>
        </div>

        <h3>Enjoy!</h3>
      </div>
    </div>
  )
}

export default Directions;