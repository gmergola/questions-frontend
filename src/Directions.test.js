import React from "react";
import { renderWithRouter } from './testHelper';
import Directions from "./Directions";


it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<Directions />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function() {
  const { asFragment } = renderWithRouter(<Directions />);
  expect(asFragment()).toMatchSnapshot();
});
