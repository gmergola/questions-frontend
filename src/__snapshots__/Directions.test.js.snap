// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Directions-container"
  >
    <button
      class="Directions-home-button"
    >
      <b>
        Back Home
      </b>
    </button>
    <div>
      <h3>
        A Family Feud inspired game with a twist!
      </h3>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Select a question
        </span>
      </div>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Once you've selected a question decide who is on team1 and who is on team 2.
        </span>
      </div>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Once you've decided on the teams, have team1 and team 2 each submit an answer. After that, the team with the highest voted answer starts!
        </span>
      </div>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Once the starting team has been selected, that team has three chances to guess as many correct answers as they can. If that team hits three incorrect answers, it is the other team's turn.
        </span>
      </div>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Every question has 6 answers, when all 6 correct answers have been guessed, the team with the highest remaining points wins that round!
        </span>
      </div>
      <h3>
        Enjoy!
      </h3>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Directions-container"
  >
    <button
      class="Directions-home-button"
    >
      <b>
        Back Home
      </b>
    </button>
    <div>
      <h3>
        A Family Feud inspired game with a twist!
      </h3>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Select a question
        </span>
      </div>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Once you've selected a question decide who is on team1 and who is on team 2.
        </span>
      </div>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Once you've decided on the teams, have team1 and team 2 each submit an answer. After that, the team with the highest voted answer starts!
        </span>
      </div>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Once the starting team has been selected, that team has three chances to guess as many correct answers as they can. If that team hits three incorrect answers, it is the other team's turn.
        </span>
      </div>
      <div
        class="Directions-bullet-point"
      >
        <span
          class="dot"
        />
        <span>
          Every question has 6 answers, when all 6 correct answers have been guessed, the team with the highest remaining points wins that round!
        </span>
      </div>
      <h3>
        Enjoy!
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
